community/kernel/src/main/java/org/neo4j/graphdb/factory/GraphDatabaseSettings.java

```diff
@@ -529,8 +529,24 @@ public String providerName()
         }
     }
 
-    @Description( "Index provider to use when creating new indexes." )
-    public static final Setting<String> default_schema_provider =
+    @Description( "Index provider to use for newly created schema indexes." +
+            "An index provider may store different value types in separate physical indexes." +
+            "lucene-1.0: Store spatial and temporal value types in native indexes, remaining value types in a Lucene index." +
+            "lucene+native-1.0: Store numbers in a native index and remaining value types like lucene-1.0." +
+            "This improves read and write performance for non-composite indexed numbers." +
+            "lucene+native-2.0: Store strings in a native index and remaining value types like lucene+native-1.0." +
+            "This improves write performance for non-composite indexed strings." +
+            "This version of the native string index has a value limit of 4047B, such that byte-representation" +
+            "of a string to index cannot be larger than that limit, or the transaction trying to index such a value will fail." +
+            "This version of the native string index also has reduced performance for CONTAINS and ENDS WITH queries," +
+            "due to resorting to index scan+filter internally." +
+
+            "Native indexes generally has these benefits over Lucene:" +
+            "- Faster writes" +
+            "- Less garbage and heap presence" +
+            "- Less CPU resources per operation" +
+            "- Controllable memory usage, due to being bound by the page cache" )
```

Comment:
There are no line endings or any other whitespace at your string concatenation points. I think this will mess up the formatting.

Reply:
Yes. How I understood it the newlines are managed by the one reading and parsing these. Although a newline _could_ work.

Reply:
The docs tools don't parse _the code_, right? They look at the string in the annotation, don't they? That's at least what I would expect. And the formatting in the resulting string is messed up, which is my complaint here. It's all one line.

Reply:
I imagined newlines being inserted wherever needed if the one reading/parsing that one-line string felt like it. I'll insert some newlines there to force this layout.